statsmodels/base/optimizer.py

```diff
@@ -217,6 +217,36 @@ def _fit_regularized(self, params):
 ########################################
 # Helper functions to fit
 
+
+def _fit_minimize(f, score, start_params, fargs, kwargs, disp=True,
+                        maxiter=100, callback=None, retall=False,
+                        full_output=True, hess=None):
+    kwargs.setdefault('min_method', None)
+
+    # prepare options dict for minimize
+    filter_opts = ['extra_fit_funcs', 'niter', 'min_method', 'tol']
+    options = {k:v for k,v in kwargs.items() if k not in filter_opts}
+    options['disp']    = disp
+    options['maxiter'] = maxiter
+
+    # we're using BFGS by default, which doesn't need hessian
+    if kwargs['min_method'] == None: hess = None
```

Comment:
to avoid the runtimewarning (based on your comment in the tests, I didn't check myself)
we could expand this to the other major "known" min_methods
e.g.
```
no_hess = ['Nelder-Mead', 'bfgs', ...]   # could also be defined outside of function, but likely not worth doing that
if kwargs['min_method'] in no_hess: 
    hess = None
```